rcljava/src/main/java/org/ros2/rcljava/events/Event.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.ros2.rcljava.events;
+
+import java.lang.ref.WeakReference;
+
+import org.ros2.rcljava.interfaces.Disposable;
+import org.ros2.rcljava.events.EventStatus;
+
+/**
+ * This class serves as a bridge between a rcl_event_t and RCLJava.
+ * An Event must be created via
+ * @{link Publisher#createEvent(Class&lt;T&gt;, Consumer&lt;T&gt;)}
+ * @{link Subscription#createEvent(Class&lt;T&gt;, Consumer&lt;T&gt;)}
+ *
+ * @param <T> The status event type.
+ * @param <ParentT> The parent class type.
+ */
+public interface Event<T extends EventStatus, ParentT extends Disposable> extends Disposable {
```

Comment:
What's the plan for integrating this with the wait set?

Reply:
That's a great question.
I wasn't sure if just adding another condition [here](https://github.com/osrf/ros2_java/blob/10cc08d53c632103965d5bebb92653a0123b99fa/rcljava/src/main/java/org/ros2/rcljava/executors/BaseExecutor.java#L76) to handle events, or if creating a `Waitable` was a better idea.

Long term, having the waitable concept is the best option.
But if we want to continue "hacking something that works" fast, I can just handle yet another custom case.

What do you think?

Reply:
I agree, long term having a Waitable interface makes sense. I would say that introducing a Waitable interface is a separate feature and is outside the scope of supporting these rcl events.